### PR TITLE
Silence "cut: write error: Broken pipe" log spew

### LINKED ISSUE #14305: Ray printing "(raylet) cut: write error: Broken pipe" constantly
Hi!

When using ray, I constantly get this message from raylet (every 1-2 secs it is printed):

```
(raylet) cut: write error: Broken pipe
```

This is the output of raylet.out (the warning part is repeated all over again)

```
[2021-02-24 13:49:00,424 I 29643 29643] io_service_pool.cc:36: IOServicePool is running with 1 io_service.
[2021-02-24 13:49:00,432 I 29643 29643] store_runner.cc:29: Allowing the Plasma store to use up to 190.873GB of memory.
[2021-02-24 13:49:00,432 I 29643 29643] store_runner.cc:42: Starting object store with directory /dev/shm and huge page support disabled
[2021-02-24 13:49:00,433 I 29643 29643] grpc_server.cc:75: ObjectManager server started, listening on port 32793.
[2021-02-24 13:49:00,436 I 29643 29643] node_manager.cc:189: Initializing NodeManager with ID 487e06036c0f9b3f34e14ef02d8feca6f064cede136a2e2fb44509dd
[2021-02-24 13:49:00,436 I 29643 29643] grpc_server.cc:75: NodeManager server started, listening on port 43854.
[2021-02-24 13:49:00,441 I 29643 29663] agent_manager.cc:76: Monitor agent process with pid 29662, register timeout 30000ms.
[2021-02-24 13:49:00,442 I 29643 29643] raylet.cc:133: Raylet of id, 487e06036c0f9b3f34e14ef02d8feca6f064cede136a2e2fb44509dd started. Raylet consists of node_manager and object_manager. node_manager address: 10.118.230.160:43854 object_manager address: 10.118.230.160:32793 hostname: 10.118.230.160
[2021-02-24 13:49:00,444 I 29643 29643] service_based_accessor.cc:559: Received notification for node id = 487e06036c0f9b3f34e14ef02d8feca6f064cede136a2e2fb44509dd, IsAlive = 1
[2021-02-24 13:49:02,918 I 29643 29653] store.cc:226: Object store current usage 8e-09 / 190.873 GB.
[2021-02-24 13:49:10,583 W 29643 29649] metric_exporter.cc:206: Export metrics to agent failed: IOError: 12: Method not found!. This won't affect Ray, but you can lose metrics from the cluster.
[2021-02-24 13:49:10,584 I 29643 29643] agent_manager.cc:32: HandleRegisterAgent, ip: 10.118.230.160, port: 48136, pid: 29662

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
WARNING:

You should always run with libnvidia-ml.so that is installed with your
NVIDIA Display Driver. By default it's installed in /usr/lib and /usr/lib64.
libnvidia-ml.so in GDK package is a stub library that is attached only for
build purposes (e.g. machine that you build your application doesn't have
to have Display Driver installed).
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
Linked to libnvidia-ml library at wrong path : /usr/lib64/libnvidia-ml.so.1
```

And in raylet.err is just the error all over again

```
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
cut: write error: Broken pipe
```

The code continues running though. It would happen anytime I use it, even with the simplest example. Any idea on how to avoid this? This is happening with Python 3.7.4 and ray 1.2.0 and 2.0.0. I have used ray 0.8.6 before without issues.

Thanks!

### PR DESCRIPTION
**Pull request by hauntsaninja**
On machines without GPUs, this can run subprocesses that spew to
stderr. Then with log_to_driver=True, we get log spew from every
single raylet. To avoid this, disable the GPU usage check on
certain errors.

Resolves #14305

### CODE REVIEW

File: dashboard/modules/reporter/reporter_agent.py [hunk 1]
```diff
@@ -197,14 +198,24 @@ def _get_cpu_percent():
...
         try:
             gpus = gpustat.new_query().gpus
         except Exception as e:
             logger.debug(f"gpustat failed to retrieve GPU information: {e}")
+
+            # gpustat calls pynvml.nvmlInit()
+            # On machines without GPUs, this can run subprocesses that spew to stderr.
+            # Then with log_to_driver=True, we get log spew from every single raylet.
+            # To avoid this, disable the GPU usage check on certain errors.
+            # https://github.com/ray-project/ray/issues/14305
```
Comment (rkooo567) [inline dashboard/modules/reporter/reporter_agent.py:215]: Can you also link to this PR?

File: dashboard/modules/reporter/reporter_agent.py [hunk 2]
```diff
@@ -30,6 +30,7 @@
 
 # Are we in a K8s pod?
 IN_KUBERNETES_POD = "KUBERNETES_SERVICE_HOST" in os.environ
+ENABLE_GPU_USAGE_CHECK = True
```
Comment (rkooo567) [inline dashboard/modules/reporter/reporter_agent.py:33]: Maybe we shouldn't use all upper case variable here? (I think this means the constant variable in our convention).

**Review by rkooo567 — COMMENTED** [review]
Can you update the PR description?

**Comment by hauntsaninja** [comment]
Thanks, addressed!

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
Looks like some tests are failing randomly.. can you try merging the latest master?

**Comment by hauntsaninja** [comment]
Looks green!

**Pull request merged**